### PR TITLE
BFD 2223: Mapping Coverage Period Start/End Dates in V2

### PR DESCRIPTION
**Pull request by JeremyIR**
**JIRA Ticket:**
[BFD-2223](https://jira.cms.gov/browse/BFD-2223)

**User Story or Bug Summary:**

Map period start and end dates in V2

---

### What Does This PR Do?

Create mapping for part A,B,D period dates in V2 via transformer

### What Should Reviewers Watch For?

are the part A,B,D period dates properly mapped and adequately tested?

If you're reviewing this PR, please check for these things in particular:

* Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [x] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [x] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [x] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [x] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist

I have gone through and verified that...:

* [x] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [x] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [x] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [x] The data dictionary has been updated with any field mapping changes, if any were made.
* [x] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [x] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [x] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [x] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

**Review by williamharty — APPROVED** [review]
endpoint test resources look good.

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/CoverageTransformerV2.java [hunk 1]
```diff
@@ -85,9 +85,9 @@ private static Coverage transformPartA(MetricRegistry metricRegistry, Beneficiar
 
     setCoverageStatus(coverage, beneficiary.getPartATerminationCode());
 
-    beneficiary
-        .getMedicareCoverageStartDate()
-        .ifPresent(value -> TransformerUtilsV2.setPeriodStart(coverage.getPeriod(), value));
+    TransformerUtilsV2.setPeriodStart(
+        coverage.getPeriod(), beneficiary.getPartACoverageStartDate());
+    TransformerUtilsV2.setPeriodEnd(coverage.getPeriod(), beneficiary.getPartACoverageEndDate());
```
Comment (lsmitchell) [inline apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/CoverageTransformerV2.java:90]: We had checks before to only add these if the values existed; do we want the same here? What happens if nulls get added here? If nothing gets output as before that's fine; but we should double check we dont print out the period with empty values or something in the final output.

File: apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/CoverageTransformerV2Test.java [hunk 2]
```diff
@@ -512,6 +512,12 @@ public void verifyRelationshipPartD() {
     verifyRelationship();
   }
 
+  @Test
```
Comment (lsmitchell) [inline apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/CoverageTransformerV2Test.java:515]: Missing docs here; i know the rest don't really have them but please add docs at least for new tests. Less for me to fix later.

**Review by lsmitchell — COMMENTED** [review]
I think we're missing a test case or two here; since these can be null, and will be for much of our data, we should have a test that ensures when the fields are passed as null the mapping handles it without issue and does not output anything for those fields in the fhir response.
